File: cpp/tests/SetMatrixZeroesTest.cpp
#include "SetMatrixZeroes.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(SetMatrixZeroes, Solution73_1) {
  auto S73_1 = Solution73_1();

  std::vector<std::vector<int>> Matrix1{{1, 1, 1}, {1, 0, 1}, {1, 1, 1}};
  const std::vector<std::vector<int>> Expected1{
      {1, 0, 1}, {0, 0, 0}, {1, 0, 1}};
  S73_1.setZeroes(Matrix1);
  EXPECT_EQ(Expected1, Matrix1);

  std::vector<std::vector<int>> Matrix2{
      {0, 1, 2, 0}, {3, 4, 5, 2}, {1, 3, 1, 5}};
  const std::vector<std::vector<int>> Expected2{
      {0, 0, 0, 0}, {0, 4, 5, 0}, {0, 3, 1, 0}};
  S73_1.setZeroes(Matrix2);
  EXPECT_EQ(Expected2, Matrix2);
}

File: java/src/test/java/io/binac/leetcode/SetMatrixZeroesTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SetMatrixZeroesTest {
    private static final SetMatrixZeroes.Solution1 SOLUTION1 = new SetMatrixZeroes.Solution1();

    private int[][] getMATRIX1() {
        return new int[][]{{1, 1, 1}, {1, 0, 1}, {1, 1, 1}};
    }

    private final int[][] EXPECTED1 = {{1, 0, 1}, {0, 0, 0}, {1, 0, 1}};

    private int[][] getMATRIX2() {
        return new int[][]{{0, 1, 2, 0}, {3, 4, 5, 2}, {1, 3, 1, 5}};
    }

    private final int[][] EXPECTED2 = {{0, 0, 0, 0}, {0, 4, 5, 0}, {0, 3, 1, 0}};

    @Test
    void test1() {
        int[][] matrix1 = getMATRIX1();
        SOLUTION1.setZeroes(matrix1);
        assertArrayEquals(EXPECTED1, matrix1);

        int[][] matrix2 = getMATRIX2();
        SOLUTION1.setZeroes(matrix2);
        assertArrayEquals(EXPECTED2, matrix2);
    }
}